src/app/core/services/auth.guard.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from "@angular/core";
+import { CanActivate } from "@angular/router";
+import { Observable } from "rxjs";
+
+import { UserService } from "./user.service";
+
+@Injectable({ providedIn: "root" })
+export class AuthGuard implements CanActivate {
+  constructor(private user: UserService) {}
```

Comment:
you can add "Service" postfix to indicate that it's a Service and avoid collision with local property